File: solutions_5648941810974720_0/C++/JustAnotherHandle/GettingTheDigits.cpp
#include <cstring>
#include <cmath>
#include <cstdio>
#include <algorithm>
#include <iostream>
#include <ctime>
#include <vector>
#include <map>
#include <set>

using namespace std;

#define pb push_back
#define sz(a) (int)a.size()
#define fs first
#define sc second

typedef long long ll;
typedef pair<int,int> ii;

int res[12], c[300], n;
char a[2005];

void solve() {
    memset(c, 0, sizeof c);
    for (int i = 0; i < n; ++i)
        c[a[i]]++;
    res[0] = c['Z'];
    res[2] = c['W'];
    res[8] = c['G'];
    res[3] = c['H'] - c['G'];
    res[6] = c['X'];
    res[7] = c['S'] - c['X'];
    res[5] = c['V'] - res[7];
    res[4] = c['R'] - c['Z'] - c['H'] + c['G'];
    res[1] = c['O'] - c['W'] - res[4] - c['Z'];
    res[9] = (n - res[0]*4 - res[1]*3 - res[2]*3 - res[3]*5 - res[4]*4 - res[5]*4 - res[6]*3 - res[7]*5 - res[8]*5) / 4;
    ////////
    for (int i = 0; i < 10; ++i)
    for (int j = 0; j < res[i]; ++j)
        printf("%d",i);
    printf("\n");
}

int main() {
	freopen("input.txt","r",stdin);
	freopen("output.txt","w",stdout);
    int t; scanf("%d\n",&t); int tmp = t;
    while (t--) {
        scanf("%s\n",a);
        n = strlen(a);
        //////////////
        printf("Case #%d: ",tmp-t);
        solve();
    }
	return 0;
}

File: solutions_5648941810974720_1/Python/Tashiqi/GettingTheDigits.py
T=int(input())
for t in range(T):
    S=input()
    l={chr(l):0 for l in range(65, 91)}
    for i in S:
        l[i]+=1
    res = ""
    res += l['Z']*"0"
    res += (l['O']-l['Z']-l['W']-l['U'])*"1"
    res += l['W']*"2"
    res += (l['H']-l['G'])*"3"
    res += l['U']*"4"
    res += (l['V']-l['S']+l['X'])*"5"
    res += l['X']*"6"
    res += (l['S']-l['X'])*"7"
    res += l['G']*"8"
    res += (l['I']-l['G']-l['X']-l['V']+l['S']-l['X'])*"9"
    print("Case #" + str(t+1) +": " +res)
